spec/duck-model/duck_spec.rb

```diff
@@ -23,6 +23,32 @@
 
 end
 
+describe Duck do
+  describe "when re-arranging ranks " do
+    context "cannot be re-arranged" do
+
+      it "fails to save" do
+        patched_duck = Duck.new(
+          :name => 'Quacky',
+          :pond => 'Shin',
+          :age_position => 1,
+        )
+        age_ranker = Duck.rankers.detect {|ranker| ranker.name == :age }
+        Duck.expects(:rankers).once.returns([age_ranker])
+        patched_duck_ranker = age_ranker.with(patched_duck)
+        age_ranker.expects(:with).once.returns(patched_duck_ranker)
+        patched_duck_ranker.expects(:current_first).returns(nil)
+        patched_duck_ranker.expects(:current_at_rank).returns(:something_truthy)
```

Comment:
This is otherwise nil. It's need to be truthy for `rearrange_ranks` to be called

Reply:
return value never seems to need to be anything but truthy since it was introduced 8280511f674e7b9468168bf9bc5ba16537477fe5

Reply:
That's true. Perhaps it was intended for future use. For it to return a boolean you'd probably want to implement a `current_at_rank?` method.

Reply:
@brendon So, the reason I'm looking at `current_at_rank` at all is because I wanted to reproduce the problem I experienced: creating a record, specifying a position, rearrange ranks failing due to current_first being nil

In order to do that, I discovered that in order for rearrange_ranks to be called in my situation, `current_at_rank` had to be truthy.  I looked at the method and determined, for mocking purposes, I didn't need to care about what it was doing or returning

I was a little curious about `current_at_rank` since I didn't see it used anywhere else or directly tested. So, I looked through the version history and found as noted above that it was introduced to replace the condition `(current_order.find do |rankable|  rankable.rank.nil? || rankable.rank == rank end)` which confirmed that its usage in `assure_unique_position` is only to return something truthy.

Interestingly, it's truthiness was replaced by `finder.except( :order ).where( ranker.column => rank ).first` where 'finder' is more or less equivalent to 'current_order'

now, why would `current_at_rank(rank)` be truthy when `current_first` is nil, when current_first is defined as `finder.first`. I'm not sure, but it could be a race condition or a query cache issue  or related to the `except` call (though I doubt it)

Reply:
I'm not sure exactly what the intended meaning of current_at_rank is, or if it changed during the refactor. I think it is that another record exists?

Reply:
Let's continue the discussion here: #149